src/main/java/com/google/sps/servlets/ImageHandlerServlet.java

```diff
@@ -31,108 +32,105 @@
+  private Entity newPlayerEntity(String imageBlobKeyString, String displayName) {
+    player = new Entity(PLAYER_QUERY_PARAMETER);
+    player.setProperty(EMAIL_PARAMETER, USER.getEmail());
+    player.setProperty(IMAGE_ID_PARAMETER, imageBlobKeyString);
+    player.setProperty(DISPLAY_NAME_PARAMETER, displayName);
+    player.setProperty(ID_PARAMETER, USER.getUserId());
+    return player;
+  }
+
+  private String createUploadedBlobKey(HttpServletRequest request, String formInputElementName) {
     Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
     List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return DEFAULT_PARAMETER;
+    }
+
     // Our form only contains a single file input, so get the first index.
     BlobKey blobKey = blobKeys.get(0);
-    return blobKey;
-  }
 
-  private String getFileUrl(BlobKey blobKey) {
-    // Use ImagesService to get a URL that points to the uploaded file.
-    ImagesService imagesService = ImagesServiceFactory.getImagesService();
-    ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
-    // To support running in Google Cloud Shell with AppEngine's dev server,
-    // we must use the relativepath to the image, rather than
-    // the path returned by imagesService which contains a host.
-    try {
-      URL url = new URL(imagesService.getServingUrl(options));
-      return url.getPath();
-    } catch (MalformedURLException e) {
-      return imagesService.getServingUrl(options);
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return DEFAULT_PARAMETER;
     }
-  }
 
-  private boolean isValidImage(String imageUrl) {
-    if (imageUrl.endsWith(JPG_PARAMETER)
-        || imageUrl.endsWith(JPEG_PARAMETER)
-        || imageUrl.endsWith(PNG_PARAMETER)) {
-      return true;
+    // Checks that the file is the right type.
+    if (!blobInfo.getContentType().contains(IMAGE_PARAMETER)) {
+      blobstoreService.delete(blobKey);
+      return CURRENT_PLAYER_FALSE_PARAMETER;
     }
-    return false;
+    return blobKey.getKeyString();
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    try {
-      response.getWriter();
-      if (isLoggedIn) {
-        Player player = new Player("", "");
-        PlayerDatabase.addPlayerToDatabase(player);
-        imageID = PlayerDatabase.getEntityImageID();
-        displayName = PlayerDatabase.getEntityDisplayName();
-        response.getWriter().println(CURRENT_PLAYER_TRUE_PARAMETER);
-      } else {
-        response.getWriter().println(CURRENT_PLAYER_FALSE_PARAMETER);
+    response.getWriter();
+    response.setContentType(CONTENT_TYPE);
```

Comment:
This is pretty complicated. Maybe split it into two methods:

if (isLoggedIn){
    handleLoggedInUser();
} else{
    handleNotLoggedInUser();
}